### PR TITLE
[histogram] make serializable and add test

### PR DESCRIPTION
**Pull request by HugoStrand**
This adds serialization support for the histogram class so that it can be sent over mpi with a test case. Please consider merging.

### CODE REVIEW

File: test/pytriqs/base/histogram_bcast.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        for pt, val in zip(pts, h.data):
+            val = int(val)
+            #print pt, val
+            if val > 0:
+                assert( val == h_ref[pt] )
+            else:
+                assert( not h_ref.has_key(pt) )
+        
+        for key, value in h_ref.items():
+            assert( value == h_ref[key] )
```
Comment (krivenko) [inline test/pytriqs/base/histogram_bcast.py]: Can this ever fail? :)

File: test/pytriqs/base/histogram_bcast.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    return hist
+
+if mpi.is_master_node():
+
+    x = np.random.randint(0, high=10, size=100)
+    
+    h_ref = py_histogram(x)
+    h = Histogram(0, 10)
+    h << x
```
[the remark this answers is not in the document]
Comment (HugoStrand) [reply test/pytriqs/base/histogram_bcast.py]: Apparently this works, I do not know why.

File: test/pytriqs/base/histogram_bcast.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+for rank in xrange(mpi.size):
+    if rank == mpi.rank:
+
+        print '-'*72
+        print 'rank =', mpi.rank
+        print 'h =\n', h
+        print 'h_ref =\n', h_ref
+
+        # -- Compare h and h_ref
+        pts = np.array([ int(h.mesh_point(idx)) for idx in xrange(len(h.data))])
```
[the remark this answers is not in the document]
Comment (HugoStrand) [reply test/pytriqs/base/histogram_bcast.py]: Yes, thanks for pointing that out 👍

File: test/pytriqs/base/histogram_bcast.py (hunk 1 above)
Comment (HugoStrand) [reply test/pytriqs/base/histogram_bcast.py]: woopsie, that should not be there. Thanks

File: test/pytriqs/base/histogram_bcast.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    
+    hist = {}
+    for val in int_vec:
+        hist[val] = 1 + hist.get(val, 0)
+
+    return hist
+
+if mpi.is_master_node():
+
+    x = np.random.randint(0, high=10, size=100)
```
[the remark this answers is not in the document]
Comment (HugoStrand) [reply test/pytriqs/base/histogram_bcast.py]: Fair enough, I have not heard that argument (I usually fix the seed, but missed it here). I have replaced that call with a fixed random sequence.

File: test/pytriqs/base/CMakeLists.txt [hunk 5]
```diff
@@ -49,4 +49,5 @@ add_python_test(gf_tensor_valued)
 
 # Histograms
 add_python_test(histograms)
+add_python_test(histogram_bcast)
```
[the remark this answers is not in the document]
Comment (HugoStrand) [reply test/pytriqs/base/CMakeLists.txt]: I do not know the cmake api for doing this. I can not find any parallel runs in the python tests, only on the c++ level.

**Comment by HugoStrand** [comment]
Dear Igor,

Thank you for the feedback, it is very helpful for me. I have tried to account for your comments and updated the pull request.

Please reconsider merging.

Best,
Hugo

File: test/pytriqs/base/histogram_bcast.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (krivenko) [reply test/pytriqs/base/histogram_bcast.py]: Okay, now I get it.
NumPy array implements `__rlshift__()` method, and here it's called as `x.__rlshift__(h)` after `h.__lshift__(x)` fails.

File: test/pytriqs/base/CMakeLists.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (krivenko) [reply test/pytriqs/base/CMakeLists.txt]: Please, try the following,
```cmake
set(TEST_MPI_NUMPROC 4)
add_python_test(histogram_bcast)
```
If this doesn't work, I'll approve this PR as is.

**Pull request merged**